src/main/scala/gestalt/Symbols.scala

```diff
@@ -15,5 +15,8 @@ trait Symbols { this: Toolbox =>
 
     /** type of a member with respect to a prefix */
     def asSeenFrom(mem: Symbol, prefix: Type): Type
+
+    /** subst symbols in tree */
+    def subst(tree: tpd.Tree)(from: List[Symbol], to: List[Symbol]): tpd.Tree
```

Comment:
I agree it's pretty low-level, doing typed transform in my mind is as difficult as compiler transform. Currently, I'm trying to see what APIs are needed to support such typed transform.

`unlift` as a macro seems don't help here. We can reflect on the design more on the problem after we've a tentative solution. My impression is that it's impossible to avoid low-level stuff in order to do correct typed transform. Otherwise, we need to restrict what kind of macros are supported.